Sub/Src/ComputerVision/Nodes/CamNode.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+import cv2
+import math
+import random
+import socket
+import sys
+import struct
+import math
+import io
+import time
+from ctypes import *
+from libs.darknet import *
+from MechOS.message_passing.Nodes.node_base import node_base
+
+class CamNode(node_base):
+
+    def __init__(self, MEM, IP):
+
+        # IP and MEM RAM locations
+        node_base.__init__(self, MEM, IP)
+
+        # Instantiations
+
+        #--MESSAGING INFO--#
+        self.MAX_UDP_PACKET_SIZE = 1500
+
+        # The end byte of the image sent over udp
+        self.END_BYTE = bytes.fromhex('c0c0')*2
+
+        # the Max Packet Size the Port will accept
+        self.MAX_PACKET_SIZE = 1500
+
+        #--CAMERA INSTANCE--#
+        self.capture = cv2.VideoCapture(1)
+
+        # Maximum image size the Tegra allows without timeout
+        self.capture.set(3, 450)
+        self.capture.set(4, 450)
+
+        # Neural Network Loaded from instance in darknet module
+        self.net  = load_net(b'./libs/darknet/cfg/yolov3-tiny.cfg',
+                       b'./libs/darknet/weights/yolov3-tiny.weights',
+                       0)
+
+        self.meta = load_meta(b'./libs/darknet/cfg/coco.data')
+
+        self.fps = self.capture.get(cv2.CAP_PROP_FPS)
+        print("Frames per second using video.get(cv2.CAP_PROP_FPS) : {0}".format(self.fps))
+
+    def run(self):
+        while(True):
+            # Capture frame-by-frame
+            ret, byte_frame = self.capture.read()
+
+            # Operations on the frame
+            '''
+            Yolo: Operations on Frame For Yolo
+            '''
+            if ret:
+                r = detect(self.net, self.meta, byte_frame)
+
+                for i in r:
+                    x, y, w, h = i[2][0], i[2][1], i[2][2], i[2][3]
+                    xmin, ymin, xmax, ymax = convertBack(float(x), float(y), float(w), float(h))
+                    pt1 = (xmin, ymin)
+                    pt2 = (xmax, ymax)
+                    cv2.rectangle(byte_frame, pt1, pt2, (0, 255, 0), 2)
+                    cv2.putText(byte_frame, i[0].decode() + " [" + str(round(i[1] * 100, 2)) + "]", (pt1[0], pt1[1] + 20), cv2.FONT_HERSHEY_SIMPLEX, 1, [0, 255, 0], 4)
```

Comment:
It looks like you are drawing the detection onto the image, however I don't see anywhere where you actually save the data and send it to be used by something else.

Reply:
Okay, we will save it to some class variable to be accessed.